lib/src/code_field/text_editing_value.dart

```diff
@@ -185,4 +185,28 @@ extension TextEditingValueExtension on TextEditingValue {
       text: text,
     );
   }
+
+  /// Checks if the newValue is most likely caused by the `delete` button.
+  bool isDelete(TextEditingValue newValue) {
```

Comment:
Make this a method on the new value.
```suggestion
  /// Checks if this could be produced by pressing delete on the [oldValue].
  bool wasDeletePressedOn(TextEditingValue oldValue) {
```